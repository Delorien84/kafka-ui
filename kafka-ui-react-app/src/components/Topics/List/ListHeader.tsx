import { TopicColumnsToSort } from 'generated-sources';
import React from 'react';

import ListHeaderCell from './ListHeaderCell';

export interface ListHeaderProps {
  orderBy: TopicColumnsToSort | null;
  setOrderBy: React.Dispatch<React.SetStateAction<TopicColumnsToSort | null>>;
}

const ListHeader: React.FC<ListHeaderProps> = ({ orderBy, setOrderBy }) => {
  return (
    <thead>
      <tr>
        <ListHeaderCell
          value={TopicColumnsToSort.NAME}
          title="Topic Name"
          orderBy={orderBy}
          setOrderBy={setOrderBy}
        />
        <ListHeaderCell
          value={TopicColumnsToSort.TOTAL_PARTITIONS}
          title="Total Partitions"
          orderBy={orderBy}
          setOrderBy={setOrderBy}
        />
        <ListHeaderCell
          value={TopicColumnsToSort.OUT_OF_SYNC_REPLICAS}
          title="Out of sync replicas"
          orderBy={orderBy}
          setOrderBy={setOrderBy}
        />
        <th>Type</th>
        <th> </th>
      </tr>
    </thead>
  );
};

export default ListHeader;
