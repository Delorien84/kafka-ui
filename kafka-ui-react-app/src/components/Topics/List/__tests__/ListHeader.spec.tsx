import ListHeader from 'components/Topics/List/ListHeader';
import { mount } from 'enzyme';
import { TopicColumnsToSort } from 'generated-sources';
import React from 'react';

describe('ListHeader', () => {
  const setOrderBy = jest.fn();
  const component = mount(
    <table>
      <ListHeader orderBy={undefined} setOrderBy={setOrderBy} />
    </table>
  );
  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('on column click', () => {
    it('calls setOrderBy', () => {
      component.find('th').at(0).simulate('click');
      expect(setOrderBy).toHaveBeenCalledTimes(1);
      expect(setOrderBy).toHaveBeenCalledWith(TopicColumnsToSort.NAME);
      component.find('th').at(1).simulate('click');
      expect(setOrderBy).toHaveBeenCalledTimes(2);
      expect(setOrderBy).toHaveBeenCalledWith(
        TopicColumnsToSort.TOTAL_PARTITIONS
      );
      component.find('th').at(2).simulate('click');
      expect(setOrderBy).toHaveBeenCalledTimes(3);
      expect(setOrderBy).toHaveBeenCalledWith(
        TopicColumnsToSort.OUT_OF_SYNC_REPLICAS
      );
    });
    it('matches the snapshot', () => {
      expect(component.find('th').at(2)).toMatchSnapshot();
    });
  });
});
