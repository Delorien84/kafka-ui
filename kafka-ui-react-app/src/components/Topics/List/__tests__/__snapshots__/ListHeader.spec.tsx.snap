// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader matches the snapshot 1`] = `
<table>
  <ListHeader
    setOrderBy={[MockFunction]}
  >
    <thead>
      <tr>
        <th
          className="is-clickable"
          onClick={[Function]}
        >
          Topic Name
           
          <span
            className="icon is-small"
          >
            <i
              className="fas fa-caret-down"
            />
          </span>
        </th>
        <th
          className="is-clickable"
          onClick={[Function]}
        >
          Total Partitions
           
          <span
            className="icon is-small"
          >
            <i
              className="fas fa-caret-down"
            />
          </span>
        </th>
        <th
          className="is-clickable"
          onClick={[Function]}
        >
          Out of sync replicas
           
          <span
            className="icon is-small"
          >
            <i
              className="fas fa-caret-down"
            />
          </span>
        </th>
        <th>
          Type
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
  </ListHeader>
</table>
`;

exports[`ListHeader on column click matches the snapshot 1`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Topic Name
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;

exports[`ListHeader on column click matches the snapshot 2`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Topic Name
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;

exports[`ListHeader on column click matches the snapshot 3`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Total Partitions
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;

exports[`ListHeader on column click matches the snapshot 4`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Total Partitions
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;

exports[`ListHeader on column click matches the snapshot 5`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Out of sync replicas
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;

exports[`ListHeader on column click matches the snapshot 6`] = `
<th
  className="is-clickable"
  onClick={[Function]}
>
  Out of sync replicas
   
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-caret-down"
    />
  </span>
</th>
`;
